### ISSUE TITLE
WITH SECURITY_ENFORCED is not supported

### ISSUE THREAD

**Issue by dangmai** [paustint/soql-parser-js#61]
Hey, this is a super cool project. I'm developing a Prettier plugin for Apex so I have a bunch of test cases that might be beneficial for this project as well. This is one of them:

- [ ] Feature
- [x] Bug Request

## Description
This query can't be parsed:

```
SELECT Id FROM Account WITH SECURITY_ENFORCED
```

Here are all the test cases I have in my library in case you want to use some for your project: https://github.com/dangmai/prettier-plugin-apex/blob/master/tests/soql/SOQLClass.cls All of those are syntactically correct.

**Comment by paustint** [comment]
Thanks for the comment! I have been following the Prettier Apex project and really look forward to it working! I am a huge prettier fan and really impressed you are taking it on.

I am using a grammar file that was created by MuleSoft, which will need to be updated to support changes in the SOQL language.

I will take a look at your test-cases and see if I can benefit from them.

In case you have not looked, I have about 55 test cases that might be helpful for you as well: https://github.com/paustint/soql-parser-js/blob/master/test/TestCases.ts

**Comment by dangmai** [comment]
Haha yup I was actually going through your test cases to see if I can use some when I realized that a few keywords are not yet supported like WITH SECURITY_ENFORCED and ALL ROWS.

Best of luck!

**Comment by paustint** [comment]
@dangmai - I finally got around to adding this feature.
FYI, here was the test case that I used to build against in case you are curious:

`SELECT Title FROM KnowledgeArticleVersion WHERE PublishStatus = 'online' WITH DATA CATEGORY Geography__c ABOVE usa__c WITH SECURITY_ENFORCED`.

Also - I am sure you know, bit just in case you don't, `WITH SECURITY_ENFORCED` is not allowed to come before `WITH DATA CATEGORY` from what I could tell.

**Comment by dangmai** [comment]
I actually did not know that! There are many hidden quirks from Salesforce that I've uncovered during my development for `prettier-plugin-apex` that I'm no longer surprised though 🤣 Their SOQL syntax [page](https://developer.salesforce.com/docs/atlas.en-us.soql_sosl.meta/soql_sosl/sforce_api_calls_soql_select.htm) helped me a lot when it comes to expression/keyword orders, but that page misses several keywords/operators that I only discovered by reverse engineering their parser.

Anyway, thanks for the fix. Good luck with the project!
